README.md

```diff
@@ -35,12 +35,79 @@ docker run -v path/to/my/repo:/repo checkmarx/2ms git /repo
 
-- `--confluence` The URL of the Confluence instance to scan.
-- `--confluence-spaces` A comma-separated list of Confluence spaces to scan.
-- `--confluence-username` confluence username or email
-- `--confluence-token` confluence token
+```
+2ms Secrets Detection: A tool to detect secrets in public websites and communication services.
+
+Usage:
+  2ms [command]
+
+Commands
+  confluence  Scan Confluence server
+  discord     Scan Discord server
+  filesystem  Scan local folder
+  git         Scan Git repository
+  paligo      Scan Paligo instance
+  slack       Scan Slack team
+
+Additional Commands:
+  completion  Generate the autocompletion script for the specified shell
+  help        Help about any command
+
+Flags:
+      --config string          YAML config file path
+  -h, --help                   help for 2ms
+      --log-level string       log level (trace, debug, info, warn, error, fatal) (default "info")
+      --regex stringArray      custom regexes to apply to the scan, must be valid Go regex
+      --report-path strings    path to generate report files. The output format will be determined by the file extension (.json, .yaml, .sarif)
+      --stdout-format string   stdout output format, available formats are: json, yaml, sarif (default "yaml")
+      --tags strings           select rules to be applied (default [all])
+  -v, --version                version for 2ms
+
+Use "2ms [command] --help" for more information about a command.
+```
+
+<!-- command-line:end -->
+
+| :warning: Using configuration env or file                                                                                                                                                      |
+| :--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Please note that even using configuration file or environment variables, you still need to specify the subcommand name in the CLI arguments. Also, positional arguments are not yet supported. |
+
+### Environment Variables
+
+To use a flag as an environment variable, see the following rules:
+
+- Replace `-` with `_`
+- Start with `2MS_`
+- Prefer uppercase
+- Append the subcommand name(s) (if any) with `_`
+
+Examples:
+
+- `--log-level` -> `2MS_LOG_LEVEL`
+- `paligo instance` -> `2MS_PALIGO_INSTANCE`
+
+### Configuration File
+
+You can use `--config` flag to specify a configuration file. The configuration file is a YAML/JSON file with the following structure:
+
+```yaml
+# global flags that will be applied to all commands
+log-level: info
+report-path:
+  - ./report.yaml
+  - ./report.json
+  - ./report.sarif
+
+# the subcommand will be selected from the CLI arguments
+# the flags below will be applied to the selected subcommand
+paligo:
+  instance: your-instance
+  username: your-username
+  # you can combine config file and Environment Variables
+  # token: your-token
+```
```

Comment:
Do you want me to document all the available options for all the plugins?

I prefer not to do it. I gave only a small example, to leave the users to create their config based on the `help` message, which will be always updated in the documentation.
If I will put here a complete example, users will copy it and things will not work as expected, because we changed the behavior but forgot to update the documentation.

Reply:
Will be implemented in #122